src/loophole.coffee

```diff
@@ -2,19 +2,35 @@ vm = require 'vm'
 
 exports.allowUnsafeEval = (fn) ->
   previousEval = global.eval
+  callback = -> global.eval = previousEval
+  error = false
   try
     global.eval = (source) -> vm.runInThisContext(source)
-    fn()
+    fn(callback)
```

Comment:
`fn(restorePreviousEval)` still needs to be `try`ed because it might throw exception and `restorePreviousEval` is never called.